diff --git a/src/transform-features.js b/src/transform-features.js
index 9cbb92ef1b46d42edcc5f9407b812a80e8d15130..d2c394721bebeef27dd3f06ad54bdfa7695dce28 100644
--- a/src/transform-features.js
+++ b/src/transform-features.js
@@ -51,7 +51,15 @@ function transformFeatures(ocadFile, createObjects, createElement, options) {
   options = {
     ...defaultOptions,
     ...options,
-    colors: ocadFile.colors,
+    colors: new Proxy(ocadFile.colors, {
+      get(target, prop, receiver) {
+        const val = Reflect.get(target, prop, receiver)
+        if (val === undefined && typeof prop === 'string' && /^\d+$/.test(prop)) {
+          return { rgb: '#000000', renderOrder: 0 }
+        }
+        return val
+      }
+    }),
     idCount: ocadFile.objects.length,
   }
 
